resources/js/src/app/components/checkout/ShippingProfileSelect.js

```diff
@@ -34,19 +26,22 @@ Vue.component("shipping-profile-select", {
          */
         onShippingProfileChange: function()
         {
-            ResourceService.getResource("checkout")
-                .set(this.checkout)
-                .done(function()
+            this.$store.dispatch("selectShippingProfile", this.shippingProfileList.find(shippingProfile => shippingProfile.parcelServiceId === shippingProfileId))
+                .then(data =>
+                {
+                    document.dispatchEvent(new CustomEvent("afterShippingProfileChanged", {detail: this.shippingProfileId}));
+                },
+                error =>
                 {
-                    document.dispatchEvent(new CustomEvent("afterShippingProfileChanged", {detail: this.checkout.shippingProfileId}));
-                }.bind(this));
+                    console.log("error");
+                });
 
             this.validate();
         },
 
         validate: function()
         {
-            this.checkoutValidation.shippingProfile.showError = !(this.checkout.shippingProfileId > 0);
+            this.$store.commit("setShippingProfileShowError", !(this.shippingProfileId > 0));
```

Comment:
wieso !(this. shippingProfileId > 0)
sieht besser aus: this. shippingProfileId === 0